bambi/plots/utils.py

```diff
@@ -12,25 +14,74 @@
-            raise TypeError("`contrast` must be a list, dict, or string")
+        """
+        Post init that sets the name and values attributes based on the
+        the effect type (i.e., slopes or comparisons), if the user provided their
+        own values, and dtype of the variable
+        """
+        if isinstance(self.variable, dict):
+            self.user_passed = True
+            self.passed_values = np.array(list(self.variable.values())[0])
+            self.values = self.passed_values
+            if self.kind == "slopes":
+                self.values = self.epsilon_difference(self.passed_values, self.eps)
+                if self.values.ndim > 1:
+                    self.values = self.values.flatten()
+            self.name = list(self.variable.keys())[0]
+        elif isinstance(self.variable, (list, str)):
+            self.name = self.variable
+            if isinstance(self.variable, list):
+                self.name = " ".join(self.variable)
+            self.values = self.set_default_variable_values()
+        elif not isinstance(self.variable, (list, dict, str)):
+            raise TypeError("`variable` must be a list, dict, or string")
+
+    def centered_difference(self, x, eps, dtype):
+        return np.array([x - eps, x + eps], dtype=dtype)
+
+    def epsilon_difference(self, x, eps):
+        return np.array([x, x + eps])
+
+    def set_default_variable_values(self):
+        terms = get_model_terms(self.model)
+        # Get default values for each variable in the model
+        for term in terms.values():
+            if hasattr(term, "components"):
+                for component in term.components:
+                    # If the component is a function call, use the argument names
+                    if isinstance(component, Call):
+                        names = [arg.name for arg in component.call.args]
+                    else:
+                        names = [component.name]
+                    for name in names:
+                        if name == self.name:
+                            predictor_data = self.model.data[name]
+                            dtype = predictor_data.dtype
+                            if component.kind == "numeric":
+                                if self.grid or self.kind == "comparisons":
+                                    predictor_data = np.mean(predictor_data)
+                                if self.kind == "slopes":
+                                    values = self.epsilon_difference(predictor_data, self.eps)
+                                elif self.kind == "comparisons":
+                                    values = self.centered_difference(
+                                        predictor_data, self.eps, dtype
+                                    )
+                            elif component.kind == "categoric":
+                                values = get_unique_levels(predictor_data)
+
+        return values
 
 
 @dataclass
```

Comment:
same here

Reply:
I don't know where to add this, but I'm seeing the following block inside ConditionalInfo

```python
        elif isinstance(self.conditional, dict):
            self.covariates = {k: listify(v) for k, v in self.conditional.items()}
            self.covariates = dict(zip(covariate_kinds, self.conditional))
            self.user_passed = True
```

isn't the first line within the `elif` not required? You save something into `self.covariates` and then you rewrite it with something different. Or is that a bug?

Reply:
You are right. The first line is not required. Just removed and ran tests Tests passed :)